hikari/impl/bot.py

```diff
@@ -624,381 +515,80 @@ async def wait_for(
-                gw_recs.session_start_limit.used,
-                gw_recs.session_start_limit.total,
-                reset_at,
-            )
-        else:
-            _LOGGER.info(
-                "will connect %s/%s shards with a max concurrency of %s -- "
-                "you have started %s/%s sessions prior to connecting (resets at %s)",
-                len(self._shard_ids),
-                self._shard_count,
-                self._max_concurrency,
-                gw_recs.session_start_limit.used,
-                gw_recs.session_start_limit.total,
-                reset_at,
-            )
-
-    def _max_concurrency_chunker(self) -> typing.Iterator[typing.Iterator[int]]:
-        """Yield generators of shard IDs.
-
-        Each yielded generator will yield every shard ID that can be started
-        at the same time.
-
-        You should then wait 5 seconds between each window.
-
-        The first window will always contain a single shard.
-        The reasoning behind this is to ensure we are able to successfully
-        connect on one shard before spamming the gateway with failed
-        connections on large applications, as this can exhaust the daily
-        identify limit.
-        """
-        n = 0
-        is_first = True
-
-        while n < self._shard_count:
-            next_window = [i for i in range(n, n + self._max_concurrency) if i in self._shard_ids]
-            # Don't yield anything if no IDs are in the given window.
-            if is_first:
-                is_first = False
-                first, next_window = next_window[0], next_window[1:]
-                yield iter((first,))
-
-            if next_window:
-                yield iter(next_window)
-
-            n += self._max_concurrency
-
-    async def _abort_shards(self) -> None:
-        """Close all shards and wait for them to terminate."""
-        for shard_id in self._shards:
-            _LOGGER.debug("stopping shard %s", shard_id)
-            await self._shards[shard_id].close()
-        await asyncio.gather(*self._tasks.values(), return_exceptions=True)
-
-    async def _gather_shard_lifecycles(self) -> None:
-        """Await all shards.
-
-        Ensure shards are requested to close before the coroutine function
-        completes.
-        """
-        _LOGGER.debug("gathering shards")
-        gatherer = asyncio.gather(*self._tasks.values())
-        waiter = asyncio.create_task(self._request_close_event.wait(), name="listen for bot closure events")
-
-        try:
-            await asyncio.wait([gatherer, waiter], return_when=asyncio.FIRST_COMPLETED)
-
-            if not waiter.done():
-                waiter.cancel()
-        finally:
-            _LOGGER.debug("gather terminated, shutting down shard(s)")
-            aborter = asyncio.shield(self.close())
+    @staticmethod
+    def _destroy_loop(loop: asyncio.AbstractEventLoop) -> None:
+        async def murder(future: asyncio.Future[typing.Any]) -> None:
```

Comment:
would this not be the "murderer" not just "murder"

Reply:
it murders a future.

Reply:
meaning it's a future murderer